Use Xcode 15.0 RC 1 in CI

mokagio opened the pull request:
## Description
What it says in the title. Building up on @Ecarrion's work in https://github.com/woocommerce/woocommerce-ios/pull/10793

## Testing instructions
If CI is green, we're green.

## Screenshots
N.A.


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mokagio committed:
Use Xcode 15.0 RC 1 in CI

spencertransier reviewed (changes requested):
@mokagio The non-RC Xcode 15 image is now available and can be used instead of `rc.1`: `xcode-15.0`

mokagio committed:
Use Xcode 15.0 in CI

Co-authored-by: Spencer Transier <spencer@transier.us>

mokagio committed:
Remove unused `taxRateID` definition

The compiler said: Value 'taxRateID' was defined but never used;
consider replacing with boolean test.

mokagio committed:
Remove unused definitions from `AuthenticationManager`

mokagio committed:
Import Networking framework in files that use the `Note` `enum`

The Xcode compiler said "Cannot use enum 'Kind' here; 'Networking' was
not imported by this file".

mokagio committed:
Use latest CI toolkit plugin (for better test annotation)

mokagio committed:
Upgrade `CountryListMapperTests` to use `try` and `XCTUnwrap`

This is to get better information about the failures in CI, see
https://buildkite.com/automattic/woocommerce-ios/builds/16314#018afd83-d608-49e2-a8db-3dddeaba68b7/6-14

iangmaia commented on Networking/NetworkingTests/Mapper/CountryListMapperTests.swift:
```diff
@@ -43,23 +37,23 @@ private extension CountryListMapperTests {
     /// Returns the CountryListMapperTests output upon receiving `filename` (Data Encoded)
     ///
-    func mapCountries(from filename: String) -> [Country]? {
+    func mapCountries(from filename: String) throws -> [Country]? {
```
Not directly related to this PR, but since the method is now throwing an exception, what about throwing an exception in the `guard`-`else`, then making it return a non-nil `[Country]` and propagating this change to other call sites?

The alternative is to stop throwing these exceptions and just `try?` instead of `try!`, but consequently missing them -- which for this test I suppose can be helpful.

iangmaia commented:
I saw that a couple of unit tests were failing on CI with a strange reason:

> XCTAssertEqual failed: ("Optional("Jun 7, 2021 at 11:59:46 AM")") is not equal to ("Optional("Jun 7, 2021 at 11:59:46 AM")")

and 

> XCTAssertEqual failed: ("["Monday, Jan 3", "Monday, Jan 3 at 1:00 AM"]") is not equal to ("["Monday, Jan 3", "Monday, Jan 3 at 1:00 AM"]")

🤷 

A few UI Tests are also failing. Not sure if this can be related to the Xcode upgrade and be worth digging into, or perhaps this branch is missing some recent trunk changes?

jaclync commented:
Hi @mokagio 👋 I'm also looking into updating Xcode to 15 on CI for HACK Week (as this is a precondition for the main HACK Week project if we ever ship it) in https://github.com/woocommerce/woocommerce-ios/pull/11352, and just saw your PR here. Just wanted to check, are you working on this for this HACK Week? I'm going to look into the unit test failures and UI test failures, then test the app on a device just to make sure.

The main app issues for Xcode 15 have been resolved based on peaMlT-fw-p2.

spencertransier commented:
Superseded by #11352
